File: generated-code/typescript/codecraft/TcpReadWrite/src/color.ts
import { Stream } from "./stream";

/**
 * RGBA Color
 */
export class Color {
    /**
     * Red component
     */
    r: number
    /**
     * Green component
     */
    g: number
    /**
     * Blue component
     */
    b: number
    /**
     * Alpha (opacity) component
     */
    a: number

    constructor(r: number, g: number, b: number, a: number) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    /**
     * Read Color from input stream
     */
    static async readFrom(stream: Stream): Promise<Color> {
        let r;
        r = await stream.readFloat();
        let g;
        g = await stream.readFloat();
        let b;
        b = await stream.readFloat();
        let a;
        a = await stream.readFloat();
        return new Color(r, g, b, a)
    }

    /**
     * Write Color to output stream
     */
    async writeTo(stream: Stream) {
        let r = this.r;
        await stream.writeFloat(r);
        let g = this.g;
        await stream.writeFloat(g);
        let b = this.b;
        await stream.writeFloat(b);
        let a = this.a;
        await stream.writeFloat(a);
    }
}

File: generated-code/python/codecraft/TcpReadWrite/color.py
from stream_wrapper import StreamWrapper

class Color:
    """RGBA Color"""

    __slots__ = ("r","g","b","a",)

    r: float
    g: float
    b: float
    a: float

    def __init__(self, r: float, g: float, b: float, a: float):
        self.r = r
        """Red component"""
        self.g = g
        """Green component"""
        self.b = b
        """Blue component"""
        self.a = a
        """Alpha (opacity) component"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Color":
        """Read Color from input stream
        """
        r = stream.read_float()
        g = stream.read_float()
        b = stream.read_float()
        a = stream.read_float()
        return Color(r, g, b, a)
    
    def write_to(self, stream: StreamWrapper):
        """Write Color to output stream
        """
        stream.write_float(self.r)
        stream.write_float(self.g)
        stream.write_float(self.b)
        stream.write_float(self.a)
    
    def __repr__(self):
        return "Color(" + \
            repr(self.r) + \
            ", " + \
            repr(self.g) + \
            ", " + \
            repr(self.b) + \
            ", " + \
            repr(self.a) + \
            ")"